pkg/node/address_linux.go

```diff
@@ -42,6 +41,11 @@ func firstGlobalAddr(intf string, preferredIP net.IP, family int, preferPublic b
 
 retryInterface:
 	addr, err := netlink.AddrList(link, family)
+	// Ensure stable order so that the same IP is chosen every time.
+	// More info in the issue GH-7637.
+	sort.SliceStable(addr, func(i, j int) bool {
+		return addr[i].LinkIndex < addr[j].LinkIndex
+	})
```

Comment:
Any reason to perform this sort before IPs are even filtered?

Reply:
I was thinking it helps with readability since we're only sorting in one place, right after fetching the addrs. I can switch back to sorting `ipsPublic` and `ipsPrivate` separately. Wdty about fetching addrs + filtering out secondary IPs + sorting in that order?

Reply:
Upside of sorting at the end is that we have less stuff to sort, though I'm not sure it makes for a significant difference.

Reply:
I'll switch this back to sorting at the end since it's not a strictly related change.